cmd/commands/subscription_test.go

```diff
@@ -0,0 +1,171 @@
+
+			err := createCommand.Execute()
+
+			Expect(err.Error()).To(HavePrefix("a DNS-1123 subdomain must consist"))
+		})
+	})
+
+	Context("when given valid args and flags", func() {
+		It("should create the subscription with the provided name", func() {
+			stdout := &strings.Builder{}
+			createCommand.SetOutput(stdout)
+			createCommand.SetArgs([]string{
+				"subscription-name", "--from", "coco-chanel", "--processor", "my-service"})
+			clientMock.On("CreateSubscription", core.CreateSubscriptionOptions{
+				Name:       "subscription-name",
+				Subscriber: "my-service",
+				Channel:    "coco-chanel",
+			}).Return(nil, nil)
+
+			err := createCommand.Execute()
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(stdout.String()).To(Equal("create completed successfully\n"))
+		})
+
+		It("should create the subscription with the service name by default", func() {
+			stdout := &strings.Builder{}
+			createCommand.SetOutput(stdout)
+			createCommand.SetArgs([]string{
+				"--from", "coco-chanel", "--processor", "my-service"})
+			clientMock.On("CreateSubscription", core.CreateSubscriptionOptions{
+				Name:       "my-service",
+				Subscriber: "my-service",
+				Channel:    "coco-chanel",
+			}).Return(nil, nil)
+
+			err := createCommand.Execute()
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(stdout.String()).To(Equal("create completed successfully\n"))
+		})
+
+		It("should create the subscription with the output channel binding", func() {
+			stdout := &strings.Builder{}
+			createCommand.SetOutput(stdout)
+			createCommand.SetArgs([]string{
+				"subscription-name", "--from", "coco-chanel", "--processor", "my-service",
+				"--to", "chanel-number-five"})
+			clientMock.On("CreateSubscription", core.CreateSubscriptionOptions{
+				Name:       "subscription-name",
+				Subscriber: "my-service",
+				Channel:    "coco-chanel",
+				ReplyTo:    "chanel-number-five",
+			}).Return(nil, nil)
+
+			err := createCommand.Execute()
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(stdout.String()).To(Equal("create completed successfully\n"))
+		})
+
+		It("should create the subscription in the output provided namespace", func() {
+			stdout := &strings.Builder{}
+			createCommand.SetOutput(stdout)
+			createCommand.SetArgs([]string{
+				"subscription-name",
+				"--from", "coco-chanel",
+				"--processor", "my-service",
+				"--to", "chanel-number-five",
+				"--namespace", "myspace"})
+			expectedOptions := core.CreateSubscriptionOptions{
+				Name:       "subscription-name",
+				Subscriber: "my-service",
+				Channel:    "coco-chanel",
+				ReplyTo:    "chanel-number-five",
+			}
+			expectedOptions.Namespace = "myspace"
+			clientMock.On("CreateSubscription", expectedOptions).Return(nil, nil)
+
+			err := createCommand.Execute()
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(stdout.String()).To(Equal("create completed successfully\n"))
+		})
+
+		It("should propagate the client error", func() {
```

Comment:
Should this say "should propagate a client error" as there won't necessarily be an error.